flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/expressions/converter/ExpressionConverter.java

```diff
@@ -135,8 +138,26 @@ public RexNode visit(ValueLiteralExpression valueLiteral) {
 				return relBuilder.getRexBuilder().makeTimeLiteral(TimeString.fromCalendarFields(
 						valueAsCalendar(extractValue(valueLiteral, java.sql.Time.class))), 0);
 			case TIMESTAMP_WITHOUT_TIME_ZONE:
-				return relBuilder.getRexBuilder().makeTimestampLiteral(TimestampString.fromCalendarFields(
-						valueAsCalendar(extractValue(valueLiteral, java.sql.Timestamp.class))), 3);
+				TimestampType timestampType = (TimestampType) type;
+				Class<?> clazz = valueLiteral.getOutputDataType().getConversionClass();
+				LocalDateTime datetime = null;
+				if (clazz == LocalDateTime.class) {
+					datetime = valueLiteral.getValueAs(LocalDateTime.class)
+						.orElseThrow(() -> new TableException("Invalid literal."));
+				} else if (clazz == Timestamp.class) {
+					datetime = valueLiteral.getValueAs(Timestamp.class)
+						.orElseThrow(() -> new TableException("Invalid literal.")).toLocalDateTime();
+				} else {
+					throw new TableException(String.format("Invalid literal of %s.", clazz.getCanonicalName()));
+				}
+				return relBuilder.getRexBuilder().makeTimestampLiteral(
```

Comment:
We could have util in `SqlDateTimeUtils` or other class: `toTimestampString(LocalDateTime)`